addons/mrp/models/product.py

```diff
@@ -195,3 +204,18 @@ def action_open_quants(self):
             res['context']['single_product'] = False
             res['context'].pop('default_product_tmpl_id', None)
         return res
+
+    def _search_qty_available_new(self, operator, value, lot_id=False, owner_id=False, package_id=False):
+        '''extending the method in stock.product to take into account kits'''
+        product_ids = super(ProductProduct, self)._search_qty_available_new(operator, value, lot_id, owner_id, package_id)
+        kit_boms = self.env['mrp.bom'].search([('type', "=", 'phantom')])
+        kit_products = self.env['product.product']
+        for kit in kit_boms:
+            if kit.product_id:
+                kit_products += kit.product_id
+            else:
+                kit_products += kit.product_tmpl_id.product_variant_ids
+        for product in kit_products:
+            if OPERATORS[operator](product.qty_available, value):
+                product_ids.append(product.id)
+        return product_ids
```

Comment:
```suggestion
        return list(set(product_ids))
```
in current implementation we could have a duplicate here, so maybe